### PR TITLE
Move Child Benefit Tax Calculator into Smart Answers

### PR DESCRIPTION
**Pull request by deborahchua**
As part of our long term vision for our GOV.UK apps which is outlined in our [RFC](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-095-long-term-future-apps.md) and the [Platform architecture principles](https://docs.google.com/document/d/1Oft4akc6dZfhhOjosNPbFpcLUOUjz7YG7QPcVZi8hww/edit#heading=h.goscl46jcc91) doc, we want to reduce the need for multiple applications where possible. The [Child benefit tax calculator](https://www.gov.uk/child-benefit-tax-calculator/main) is the only service provided by [Calculators](https://github.com/alphagov/calculators) - all the other calculators are provided by Smart Answers. It makes sense to move this service into Smart Answers and therefore retire Calculators.

The main changes in moving the calculator across to smart-answers are:
- the start page is now handled by smart answers
- each question is now following the smart-answers format of having a single question per page (from `5` questions we now have `12` questions
- where questions were previously optional we now ask the user with a `yes/no` question if they would like to provide the value
- most of the validation is handled by the [question types](https://github.com/alphagov/smart-answers/blob/master/doc/smart-answers/question-types.md)
- it no longer uses javascript to populate the number of start and stop date questions for each part year child
- some of the input types have been changed to improve accessibility as well as follow the conventions used in smart answers
- the results information is shown on a separate page, whereas in `calculators` all the questions and results are shown on a single page
- the values entered so far for each question is stored in the question summary at the bottom of the page

Calculator start page (before/after):
Screenshot 2020-08-11 at 17 15 11

Question 1 (before/after):
Screenshot 2020-08-12 at 08 55 43

Question 2 (before/after):
Screenshot 2020-08-11 at 17 19 43

Question 3 (before/after):
Screenshot 2020-08-11 at 17 28 36

Screenshot 2020-08-11 at 17 26 49

Screenshot 2020-08-11 at 17 37 03

Question 4 (before/after):
Screenshot 2020-08-11 at 17 40 06

Question 5 (before):
Screenshot 2020-08-11 at 17 07 33

Question 5 (after):
Screenshot 2020-08-11 at 17 46 07

Screenshot 2020-08-11 at 17 50 01

Results (before/after):
Screenshot 2020-08-11 at 17 54 09

Question summary (at the bottom of the page):
Screenshot 2020-08-11 at 17 51 32

Paired with @beccapearce 

Trello card - https://trello.com/c/Jh6zC0bR/2035-move-child-benefit-tax-calculator-cbtc-into-smart-answers%F0%9F%8D%90

:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/smartanswers), after merging.

### CODE REVIEW

File: lib/smart_answer_flows/child-benefit-tax-calculator.rb
Comment (thomasleese) [inline lib/smart_answer_flows/child-benefit-tax-calculator.rb:5]:
```suggestion
      start_page_content_id "0e1de8f1-9909-4e45-a6a3-bffe95470275"
```

I think it would be good if [this matches the current `content_id`](https://github.com/alphagov/publisher/pull/629/files#diff-d75e054d219dcab61db1959fca8b4f9fR25) so we keep the history in the Publishing API. It's currently published by Publisher, but it'll end up being published by Smart Answers once this is merged in.

Comment (deborahchua) [reply lib/smart_answer_flows/child-benefit-tax-calculator.rb]: Thanks @thomasleese ! We have made this change in https://github.com/alphagov/smart-answers/pull/4762/commits/d324d7156b06b8b1114f68e6bd3c62241053169e

**Comment by theseanything** [comment]
@deborahchua Is there any update on this?

**Comment by deborahchua** [comment]
@theseanything , we'e just finished making the final changes and it's now with the dept for a re-review - see https://govuk.zendesk.com/agent/tickets/4075405

**Comment by deborahchua** [comment]
This needs a technical re-review from https://github.com/alphagov/smart-answers/pull/4762/commits/f0a9e080711cb8552ed773e17fea8060b6f328ac onwards.

**Review by benjamineskola — APPROVED** [review]

**Pull request merged**
